storage/storage.go

```diff
@@ -31,22 +32,202 @@ func NewPaginationOptions(ps int32, contToken string) PaginationOptions {
+		}
+	}
+}
+
+func (u *uniqueObjectIterator) Stop() {
+	u.iter1.Stop()
+	u.iter2.Stop()
+}
+
+type combinedIterator[T any] struct {
+	iter1, iter2 Iterator[T]
+}
+
+func (c *combinedIterator[T]) Next() (T, error) {
+	val, err := c.iter1.Next()
+	if err != nil {
+		if !errors.Is(err, ErrIteratorDone) {
+			return val, err
+		}
+	} else {
+		return val, nil
+	}
+
+	val, err = c.iter2.Next()
+	if err != nil {
+		if !errors.Is(err, ErrIteratorDone) {
+			return val, err
+		}
+	}
+
+	return val, err
+}
+
+func (c *combinedIterator[T]) Stop() {
+	c.iter1.Stop()
+	c.iter2.Stop()
+}
+
+// NewCombinedIterator takes two generic iterators of a given type T and combines them into a single iterator that yields
+// all of the values from both iterators. If the two iterators yield the same value then duplicates will be returned.
+func NewCombinedIterator[T any](iter1, iter2 Iterator[T]) Iterator[T] {
+	return &combinedIterator[T]{iter1, iter2}
+}
+
+// NewStaticTupleKeyIterator returns a TupleKeyIterator that iterates over the provided slice.
+func NewStaticTupleKeyIterator(tupleKeys []*openfgapb.TupleKey) TupleKeyIterator {
+	iter := &staticIterator[*openfgapb.TupleKey]{
+		items: tupleKeys,
+	}
+
+	return iter
+}
+
+type tupleKeyIterator struct {
+	iter TupleIterator
+}
+
+var _ TupleKeyIterator = (*tupleKeyIterator)(nil)
+
+func (t *tupleKeyIterator) Next() (*openfgapb.TupleKey, error) {
+	tuple, err := t.iter.Next()
+	return tuple.GetKey(), err
+}
+
+func (t *tupleKeyIterator) Stop() {
+	t.iter.Stop()
+}
+
+// NewTupleKeyIteratorFromTupleIterator takes a TupleIterator and yields all of the TupleKeys from it as a TupleKeyIterator.
+func NewTupleKeyIteratorFromTupleIterator(iter TupleIterator) TupleKeyIterator {
+	return &tupleKeyIterator{iter}
 }
 
+// NewTupleKeyObjectIterator returns an ObjectIterator that iterates over the objects
+// contained in the provided list of TupleKeys.
+func NewTupleKeyObjectIterator(tupleKeys []*openfgapb.TupleKey) ObjectIterator {
+
+	objects := make([]*openfgapb.Object, 0, len(tupleKeys))
+	for _, tk := range tupleKeys {
+		objectType, objectID := tuple.SplitObject(tk.GetObject())
+		objects = append(objects, &openfgapb.Object{Type: objectType, Id: objectID})
+	}
+
+	return NewStaticObjectIterator(objects)
+}
+
+type staticIterator[T any] struct {
+	items []T
+}
+
+func (s *staticIterator[T]) Next() (T, error) {
+	var val T
+	if len(s.items) == 0 {
+		return val, ErrIteratorDone
+	}
+
+	next, rest := s.items[0], s.items[1:]
+	s.items = rest
+
+	return next, nil
+}
+
+func (s *staticIterator[T]) Stop() {}
+
+type staticObjectIterator struct {
+	objects []*openfgapb.Object
+}
+
+var _ ObjectIterator = (*staticObjectIterator)(nil)
+
+// NewStaticObjectIterator returns an ObjectIterator that iterates over the provided slice of objects.
+func NewStaticObjectIterator(objects []*openfgapb.Object) ObjectIterator {
+	return &staticObjectIterator{objects}
+}
+
+func (s *staticObjectIterator) Next() (*openfgapb.Object, error) {
```

Comment:
Why can't this be implemented with the `staticIterator`?

Reply:
It totally can. Good catch! That was remnant before the generic staticIterator that I added.